src/v/pandaproxy/sharded_client_cache.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/sharded_client_cache.h"
+
+#include "ssx/future-util.h"
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy {
+
+static constexpr auto clean_timer_period = 10s;
+
+ss::future<>
+sharded_client_cache::start(ss::smp_service_group sg, YAML::Node const& cfg) {
+    _smp_opts = ss::smp_submit_to_options{sg};
+
+    _clean_timer.set_callback([this] {
+        ssx::spawn_with_gate(_gate, [this] {
+            return _cache
+              .invoke_on_all(
+                _smp_opts,
+                [](kafka_client_cache& cache) {
+                    return cache.clean_stale_clients();
+                })
+              .finally([this] { _clean_timer.arm(clean_timer_period); });
+        });
+    });
+
+    _clean_timer.arm(clean_timer_period);
+
+    auto get_max_size = []() {
+        return rest::shard_local_cfg().client_cache_max_size.bind();
+    };
+
+    auto get_keep_alive = []() {
+        return rest::shard_local_cfg().client_keep_alive.bind();
+    };
```

Comment:
I think these will be defaults; the thread locals are never set. In general, globals should be avoided.